xtask/src/main.rs

```diff
@@ -0,0 +1,107 @@
+//! Custom build system for the Yorick meta-tracer.
+//!
+//! This is required because we need to separately compile parts of the codebase with different
+//! configurations.
+//!
+//!  - Interpreter code, that we expect to be able to trace, needs to be compiled with `-C
+//!    tracer=hw`. Crucially, since hardware tracing relies on blocks not being reordered by LLVM,
+//!    this flag disables optimisation.
+//!
+//!  - Anything that we don't need to trace, for example the JIT internals themselves, can (and
+//!    should) be optimised.
+//!
+//! Note that the above definitions also extend to dependencies, and this means that we can have
+//! two copies of any given crate but compiled with different flags. This is good, as it means that
+//! the dependencies of interpreter code can be traced, whereas dependencies of the JIT internals
+//! can be optimised.
+//!
+//! To this end, we have 2 Rust workspaces in the repo: the "internal" (optimised) workspace, and
+//! the "external" (unoptimised) workspace. The external workspace then talks to the internal
+//! workspace via `extern` functions defined in the ykshim crate.
+//!
+//! There are a few of implementation details to note:
```

Comment:
@bjorn3 would you be able to check I got these details right please?

Reply:
Looks correct.